### ISSUE TITLE
Not refreshing layout when switching between ARM configuration and RISC-V configuration inside single project.

### ISSUE THREAD

**Issue by AntonKrug** [gnu-mcu-eclipse/eclipse-plugins#279]
### Description

A project which has to target 2 platforms shows strange behaviour when switching configurations.

### Steps to Reproduce

1. Import to RISCV project a configuration from ARM project so you could be targeting both platforms by quickly switching.
2. In project settings now the RISCV is showing as RISCV.
3. By clicking on the top to switch configuration, now I can switch to ARM and the layout updates,
4. Switching back to RISCV project will still keep the ARM layout and not update to RISCV.

**Expected behaviour:** 

Switching between configuration should always update to the corresponding target,

**Actual behaviour:**

![1-riscv-configuration-is-riscv](https://user-images.githubusercontent.com/30338596/35818723-adf2b68a-0a98-11e8-8072-9220e3508976.PNG)
![2-arm-configuration-is-arm](https://user-images.githubusercontent.com/30338596/35818724-ae11e8a2-0a98-11e8-9922-159602087148.PNG)
![3-riscv-configuration-is-now-arm](https://user-images.githubusercontent.com/30338596/35818725-ae292da0-0a98-11e8-9c18-0ec8bfbbbb22.PNG)

### Versions

* [plug-in version] 2.6.1
* [Eclipse version] 4.7.2
* [Java version] 9.0.4
* [operating system] Windows 10 64 bit
* [toolchain version] riscv 7.2.0, arm 7.2.1

**Comment by ilg-ul** [comment]
@AntonKrug, I'm not able to reproduce this problem.

I tried with the latest version of the plug-ins and also with an older version.

I creted a new RISC-V project and imported the Debug configuration from an ARM project, saved the new configuration and then entered again in the Properties windows. Switching from ARM to RISC-V and back works as expected.

I think the step to save everything after adding the new configuration is mandatory, otherwise Eclipse might not be aware of the new configuration.

**Comment by ilg-ul** [comment]
If you can reproduce this, please come back with more details. Otherwise please close this issue.

**Comment by ilg-ul** [comment]
Close the Properties page and enter again.

**Comment by AntonKrug** [comment]
Ah, yes been doing that and even closing and opening eclipse doesn't help. Importing ARM to RISCV and vice versa is not fixing it as well.

**Comment by ilg-ul** [comment]
Does it happen on Linux too?

**Comment by AntonKrug** [comment]
No Linux works fine even without the "save".

**Comment by ilg-ul** [comment]
Then I doubt it is a problem with the plug-in.

**Comment by AntonKrug** [comment]
Can you replicate it on Windows?

**Comment by ilg-ul** [comment]
Yes. But I have no idea how to address this. As you know, the Tool Settings tab is managed by CDT, based on the toolchain definitions in plugin.xml; I have no control on the redraw logic there.

**Comment by ilg-ul** [comment]
If you can install other regular (non cross) toolchains on your windows, do the same test, and if the behaviour is the same, report it to CDT.
